react/week1/masi-todo-app/src/components/todoList.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from "react";
+import ListItem from "./listItem";
+
+const initialMockedTodoes = [
+  { id: 1, title: "Get out of bed", date: new Date("09-13-2017"), done: true },
+  { id: 2, title: "Brush teeth", date: new Date("09-14-2017") },
+  { id: 3, title: "Eat breakfast", date: new Date("09-15-2017") }
+];
+
+const MyTodoList = () => {
+  const [todoList, setTodoList] = useState(initialMockedTodoes);
+  const changeState = id => {
+    let index = todoList.findIndex(todo => todo.id === id);
+    let arrayAheadOfTarget = todoList.slice(0, index);
+    let target = Object.assign({}, todoList[index], {
+      done: !todoList[index].done
+    });
+    let arrayAfterTarget = todoList.slice(index + 1);
+    let out = [...arrayAheadOfTarget, target, ...arrayAfterTarget];
+    setTodoList(out);
+  };
```

Comment:
I am not sure if you have worked with maps but you could definitely use in this code. All this code could be replaced by:
```jsx
  const changeState = id => {
    setTodoList(currentList => currentList.map(item => {
      if(item.id === id) {
       return ({...item, done: !item.done )
      }
      return item;
    } ))
  };
```
or even further using a ternary expression:

```jsx
const changeState = id => {
    setTodoList(currentList => currentList.map(item => item.id !== id ? item : ({
        ...item,
        done: !item.done
    }) ))
  };
```

Reply:
very good point, I fixed it, and went with your first suggestion :)